glean-core/src/metrics/boolean.rs

```diff
@@ -16,6 +16,26 @@ pub struct BooleanMetric {
     meta: CommonMetricData,
 }
 
+/// A description for the `BooleanMetric` type.
+pub trait Boolean {
+    /// Creates a new instance of the metric.
+    fn new(meta: CommonMetricData) -> Self;
```

Comment:
Do we really need to expose `new` as a trait method? This is not necessarily "metric type behaviour"

Reply:
You're right, we don't need it.

Reply:
Thinking of this again, one reason to define `new`: think of [custom distributions](https://searchfox.org/glean/rev/02b67f683104ceff34c24936525f496195b3abbe/glean-core/src/metrics/custom_distribution.rs#46-51) for example, they expect a specific set of parameters to be passed. If we bubbled up these expectations to the trait, then the "glean_parser" would always know what to expect, because we'd define this on the trait.

On the other hand, we control the glean_parser. So it's not too much of a big deal.